tests/includes/juju.sh

```diff
@@ -188,15 +218,31 @@ juju_bootstrap() {
 		;;
 	esac
 
+	image_metadata=""
+
+	if [[ ${BOOTSTRAP_PROVIDER} == "vsphere" ]]; then
+		echo "====> Creating image simplestream metadata for juju ($(green "${version}:${cloud}"))"
+
+	  image_streams_dir=/tmp/image-streams
+	  setup_vsphere_simplestreams ${image_streams_dir} ${BOOTSTRAP_SERIES}
+	  image_metadata="--metadata-source ${image_streams_dir}"
+	fi
+
+
 	# When double quotes are added to ${series}, the juju bootstrap
 	# command looks correct, and works outside of the harness, but
 	# does not run, goes directly to cleanup.
 	#shellcheck disable=SC2086
 	juju bootstrap ${series} \
 		--build-agent=${BUILD_AGENT} \
-		"${provider}" "${name}" -d "${model}" "$@" 2>&1 | OUTPUT "${output}"
+		"${cloud}" "${name}" -d "${model}" ${image_metadata} "$@" 2>&1 | OUTPUT "${output}"
 
 	echo "${name}" >>"${TEST_DIR}/jujus"
+
+	if [[ ${BOOTSTRAP_PROVIDER} == "vsphere" ]]; then
```

Comment:
It would be nice if we had a more generic section for this something like
```
post_bootstrap_initialization() {
switch ${BOOTSTRAP_PROVIDER} {
}
```

I say that because there is a particular type of operation we are trying to do, and it would be good to capture that rather than slowly accreting more and more if blocks.